org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/CompositeBean.java

```diff
@@ -0,0 +1,98 @@
+package org.eclipse.scanning.api.database;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * 
+ * This class is a composite of serveral beans which
+ * should be upserted/updated/inserted using the 
+ * database service. It is used to create single requests
+ * which do multiple actions. This allows queues of 
+ * composite operations to be created with each one being
+ * a single action in terms of what the ISPyB Client submits
+ * to the database.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class CompositeBean {
+
+	private Map<Operation, Collection<Object>> data;
+    
+    public CompositeBean() {
+    	data = new LinkedHashMap<>(); // Order matters in the collection of items
+    }
+    
+    public CompositeBean(Operation type, Object... beans) {
+    	this();
+    	assertType(type);
+     	data.put(type, new ArrayList<>(Arrays.asList(beans)));
+    }
+   
+    private void assertType(Operation type) {
+       	if (!Operation.ends().contains(type)) {
+       		throw new IllegalArgumentException("Each operation set added to the composite must not be a composite. Allowed types are "+Operation.ends());
+       	}
+	}
+
+	public void add(Operation type, Object bean) {
+    	
+    	assertType(type);
+    	Collection<Object> beans = data.get(type);
+    	if (beans==null) {
+    		beans = new ArrayList<>(7); // Order matters in the collection of items
+    		data.put(type, beans);
+    	}
+    	beans.add(bean);
+    }
+    
+    public Collection<Object> set(Operation type, Object... beans) {
+       	assertType(type);
+    	return data.put(type, new ArrayList<>(Arrays.asList(beans)));
+    }
+
+	public Collection<?> get(Operation type) {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1452)